src/main/java/org/testng/DataProviderInformation.java

```diff
@@ -0,0 +1,33 @@
+package org.testng;
+
+import java.lang.reflect.Method;
+
+/**
+ * Represents information pertaining to a data provider that would be invoked.
+ */
+public class DataProviderInformation {
+    private Object instance;
+    private Method method;
+
+    public DataProviderInformation(Method method, Object instance) {
```

Comment:
@juherr - @cbeust had suggested that we don't need to be duplicating things which can be retrieved via the `Method` object using reflection. That is why I chose to discard it.

`DataProviderHolder` is internal and its being consumed by the codebase. That was why I chose to leave it as is, instead of introducing changes around it.